lib/bolt/boltdir.rb

```diff
@@ -12,13 +12,21 @@ def self.default_boltdir
       Boltdir.new(File.join('~', '.puppetlabs', 'bolt'))
     end
 
+    # Search recursively up the directory hierarchy for the Boltdir. Look for a
+    # directory called Boltdir or a file called bolt.yaml (for a control repo
+    # type Boltdir). Otherwise, repeat the check on each directory up the
+    # hierarchy, falling back to the default if we reach the root.
     def self.find_boltdir(dir)
-      local_boltdir = Pathname.new(dir).ascend do |path|
```

Comment:
Why'd you switch from `ascend`?

Reply:
It took me a while to wrap my brain around the logic and this was just where I landed. 😄 And it seemed a bit cleaner with multiple conditionals.

The ascend version would be something like this:

```rb
def self.find_boltdir(dir)
  local_boltdir = Pathname.new(dir).ascend do |path|
    if (path + BOLTDIR_NAME).directory?
      break new(path + BOLTDIR_NAME)
    elsif (path + 'bolt.yaml').file?
      break new(path)
    end
  end
                                                     
  local_boltdir || default_boltdir
end
```

Or this:

```rb
def self.find_boltdir(dir)
  Pathname.new(dir).ascend do |path|
    if (path + BOLTDIR_NAME).directory?
      return new(path + BOLTDIR_NAME)
    elsif (path + 'bolt.yaml').file?
      return new(path)
    end
  end
                                                     
  default_boltdir
end
```